torch/csrc/autograd/python_variable.cpp

```diff
@@ -154,6 +154,13 @@ static bool check_has_torch_dispatch(PyObject *obj) {
   );
 }
 
+static PyObject* getXLATensorClass() {
+  // TODO: What should we do if XLATensor doesn't exist? Return THPVariableClass?
+  // TODO: is THPObjectPtr the right pointer container?
+  static THPObjectPtr xla_tensor_ptr(py::globals()["XLATensor"].ptr());
+  return xla_tensor_ptr.get();
```

Comment:
You want to use release here otherwise xla_tensor_ptr might get de-allocated? (even though is is super unlikely as it is a type)
And the caller is then responsible for getting rid of that reference.

If you want to go yolo and just no do any refcount at all then you should use plain PyObjects here.

Reply:
hehe, @ezyang says we should go yolo, we are doing that with `THPVariableClass` anyways, it seems  😺

Reply:
The tp_alloc/tp_dealloc do the proper incref/decref.
So the only problem thst can happen is that another thread frees the type between here and the newWithVar call below. I agree that this is unlikely enough :p